### PR TITLE
Add a coverage plugin that handles ipykernel cells

### PR DESCRIPTION
**Pull request by chmp**
Open issues:

- [x] how to inject the coverage plugin with a coveragerc file?
  - document the standard way (using .coveragerc)
- [x] how to get coverage to not ignore the (non-existing) kernel directory?
  - For cell files of such as `C:/Users/{USER}/AppData/Local/Temp/ipykernel_1300/3920661193.py`, `C:/Users/{USER}/AppData/Local/Temp` works, `C:/Users/{USER}/AppData/Local/Temp/ipykernel_1300` does not
  - theory this behavior is triggered by [this line](https://github.com/nedbat/coveragepy/blob/d9156677a9450e1d613ffa25c87d6b063c28e62e/coverage/inorout.py#L194)
  - **A**: without explicit source, this issue does not come up
- [x] Another issue is that `%%ipytest --cov` gets completed to `%%ipytest --cov {MODULE}`. Therefore `{MODULE}` is added to the sources. Using `%%ipytest --cov -vv` works correctly.
   - **A**: insert a `--` in the defopts to make sure the inserted module is never parsed an argument value

Notes:

- `ipykernel` uses a custom caching compiler ([source](https://github.com/ipython/ipykernel/blob/a7d66ae2197e0d7471ba160542cf5ff7713084b5/ipykernel/ipkernel.py#L127))
- this caching compiler uses a file of the format `{TMP_PATH}/ipykernel_{pid}/{code_hash}.py` ([source](https://github.com/ipython/ipykernel/blob/a7d66ae2197e0d7471ba160542cf5ff7713084b5/ipykernel/compiler.py#L86))

### CODE REVIEW

File: ipytest/cov.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+```ini
+[run]
+plugins =
+    ipytest.cov
+```
+
+With this config file, the coverage can be collected using
+[pytest-cov][ipytest-cov-pytest-cov] with
+
+```pyhton
```
Comment (westurner) [inline ipytest/cov.py:16]: s/pyhton/python/

File: ipytest/cov.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+The plugin must be enabled in a `.coveragerc` next to the current notebook or
+the `pyproject.toml` file. See the [coverage.py docs][coverage-py-config-docs]
+for details. In case of a `.coveragerc` file, the minimal configuration reads:
+
+```ini
+[run]
+plugins =
+    ipytest.cov
+```
+
```
Comment (westurner) [inline ipytest/cov.py:12]: The `%%file` cell magic can be used to create files like `.coveragerc` from a notebook.
```
%%file .coveragerc
[run]
plugins =
    ipytest.cov
```

Would an `ipytest.generate_coveragerc()` or similar be too much? or Just `autoconfig(coverage=True)` or something?

**Review by westurner — COMMENTED** [review]
Thanks. Line and Branch Coverage help check code quality.

Common misconception: if something has 100% test coverage, it's also fuzzed.

also, re: hypothesis *property testing* and pytest-cov: https://hypothesis.readthedocs.io/en/latest/strategies.html#interaction-with-pytest-cov

File: ipytest/cov.py (hunk 2 above)
Comment (chmp) [reply ipytest/cov.py:12]: TBH. I would rather prefer, that there is an option to configure pytest-cov programatically and inject the plugin in that way. I will open an issue for further discussion.

**Pull request merged**